### ISSUE TITLE
Compatibility with Zabbix 6.2

### ISSUE THREAD

**Issue by BGmot** [ansible-collections/community.zabbix#754]
##### SUMMARY
All modules need to be compatible with recently released Zabbix 6.2

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
Two modules fail integration tests against Zabbix 6.2:
- zabbix_authentication
- zabbix_host_info

##### ANSIBLE VERSION
```
ansible 2.10.17
  config file = None
  configured module search path = ['/home/ey/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ey/env_py39_ans210/lib/python3.9/site-packages/ansible
  executable location = /home/ey/env_py39_ans210/bin/ansible
  python version = 3.9.5 (default, Nov 23 2021, 15:27:38) [GCC 9.3.0]
```

##### OS / ENVIRONMENT / Zabbix Version
Zabbix 6.2.

##### STEPS TO REPRODUCE
```
ansible-test integration -v --color --diff --python 3.9 --requirements
```

##### EXPECTED RESULTS
All tests should pass.

##### ACTUAL RESULTS
For zabbix_authentication module getting following error:
```
TASK [test_zabbix_authentication : test - update all authentication setting] ***************************************************************************************************************************************
fatal: [testhost]: FAILED! => {"changed": false, "msg": "Failed to update authentication setting, Exception: ('Error -32602: Invalid params., Invalid parameter \"/\": unexpected parameter \"ldap_host\". while sending {\"jsonrpc\": \"2.0\", \"method\": \"authentication.update\", \"params\": {\"authentication_type\": \"0\", \"http_auth_enabled\": \"1\", \"http_login_form\": \"0\", \"http_strip_domains\": \"comp,any\", \"http_case_sensitive\": \"1\", \"ldap_configured\": \"1\", \"ldap_host\": \"ldap://localhost\", \"ldap_port\": \"389\", \"ldap_base_dn\": \"ou=Users,ou=system\", \"ldap_search_attribute\": \"uid\", \"ldap_bind_dn\": \"uid=ldap_search,ou=system\", \"ldap_case_sensitive\": \"1\", \"ldap_bind_********\": \"********\", \"saml_auth_enabled\": \"1\", \"saml_sso_url\": \"https://localhost/SAML2/SSO\", \"saml_slo_url\": \"https://localhost/SAML2/SLO\", \"saml_username_attribute\": \"uid\", \"saml_sp_entityid\": \"https://localhost\", \"saml_nameid_format\": \"urn:oasis:names:tc:SAML:2.0:nameid-format:entity\", \"saml_sign_messages\": \"1\", \"saml_sign_assertions\": \"1\", \"saml_sign_authn_requests\": \"1\", \"saml_sign_logout_requests\": \"1\", \"saml_sign_logout_responses\": \"1\", \"saml_encrypt_nameid\": \"1\", \"saml_encrypt_assertions\": \"1\", \"saml_case_sensitive\": \"1\", \"passwd_min_length\": \"70\", \"passwd_check_rules\": \"15\"}, \"auth\": \"80efc2870ecbc31a9e2b80374cc0a5f3\", \"id\": 3}', -32602)"}

```

For zabbix_host_info module getting following error:
```
TASK [test_zabbix_host_info : test - Prepare host for zabbix_host_info module] *************************************************************************************************************************************
[WARNING]: The value "1" (type int) was converted to "'1'" (type string). If this does not look like what you expect, quote the entire value to ensure it does not change.
[WARNING]: The value "10050" (type int) was converted to "'10050'" (type string). If this does not look like what you expect, quote the entire value to ensure it does not change.
ok: [testhost] => {"changed": false}

TASK [test_zabbix_host_info : assert] ******************************************************************************************************************************************************************************
fatal: [testhost]: FAILED! => {
    "assertion": "prepare_host_result.changed is sameas true",
    "changed": false,
    "evaluated_to": false,
    "msg": "Assertion failed"
}
```

**Comment by BGmot** [comment]
Ability to configure several LDAP servers was introduced in Zabbix 6.2 thus zabbix_user_directory module needs to be created for zabbix_authentication to work properly with.

**Comment by mu1f407** [comment]
Hi @BGmot, the message you've got from zabbix_host_info test is not the real issue. You just didn't had your environment clean before tests and host already existed. I've sent a PR which solves failing tests for this module in Zabbix 6.2.

It's great you started working on 6.2 compatiblity. Thank you! The breaking change in authentication module is fairly extensive. I've looked a little bit at it before, but didn't had time to work on it. Let me know if you want any help.

**Comment by BGmot** [comment]
Hi @mu1f407, you are right - I was doing tests module by module and probably repeated zabbix_host_info two times and at that point did not analyze any errors, was concentrating on a list of what needs to be fixed for 6.2.
Indeed on 6.2 I get this groups returned:
```
"groups": [{"flags": "0", "groupid": "7", "name": "Hypervisors", "uuid": "1b837a3c078647049a0c00c61b4d57b5"}, {"flags": "0", "groupid": "2", "name": "Linux servers", "uuid": "dc579cd7a1a34222933f24f52a68bcd8"}]
```
I tested - your PR fixes it so I vote for it. Thanks!
